plugins/partiql-plugin/src/main/kotlin/org/partiql/plugin/PartiQLPlugin.kt

```diff
@@ -0,0 +1,493 @@
+        Fn_DATE_ADD_MINUTE__INT_TIMESTAMP__TIMESTAMP,
+        Fn_DATE_ADD_SECOND__INT32_DATE__DATE,
+        Fn_DATE_ADD_SECOND__INT64_DATE__DATE,
+        Fn_DATE_ADD_SECOND__INT_DATE__DATE,
+        Fn_DATE_ADD_SECOND__INT32_TIME__TIME,
+        Fn_DATE_ADD_SECOND__INT64_TIME__TIME,
+        Fn_DATE_ADD_SECOND__INT_TIME__TIME,
+        Fn_DATE_ADD_SECOND__INT32_TIMESTAMP__TIMESTAMP,
+        Fn_DATE_ADD_SECOND__INT64_TIMESTAMP__TIMESTAMP,
+        Fn_DATE_ADD_SECOND__INT_TIMESTAMP__TIMESTAMP,
+        Fn_DATE_DIFF_YEAR__DATE_DATE__INT64,
+        Fn_DATE_DIFF_YEAR__TIME_TIME__INT64,
+        Fn_DATE_DIFF_YEAR__TIMESTAMP_TIMESTAMP__INT64,
+        Fn_DATE_DIFF_MONTH__DATE_DATE__INT64,
+        Fn_DATE_DIFF_MONTH__TIME_TIME__INT64,
+        Fn_DATE_DIFF_MONTH__TIMESTAMP_TIMESTAMP__INT64,
+        Fn_DATE_DIFF_DAY__DATE_DATE__INT64,
+        Fn_DATE_DIFF_DAY__TIME_TIME__INT64,
+        Fn_DATE_DIFF_DAY__TIMESTAMP_TIMESTAMP__INT64,
+        Fn_DATE_DIFF_HOUR__DATE_DATE__INT64,
+        Fn_DATE_DIFF_HOUR__TIME_TIME__INT64,
+        Fn_DATE_DIFF_HOUR__TIMESTAMP_TIMESTAMP__INT64,
+        Fn_DATE_DIFF_MINUTE__DATE_DATE__INT64,
+        Fn_DATE_DIFF_MINUTE__TIME_TIME__INT64,
+        Fn_DATE_DIFF_MINUTE__TIMESTAMP_TIMESTAMP__INT64,
+        Fn_DATE_DIFF_SECOND__DATE_DATE__INT64,
+        Fn_DATE_DIFF_SECOND__TIME_TIME__INT64,
+        Fn_DATE_DIFF_SECOND__TIMESTAMP_TIMESTAMP__INT64,
+        Fn_CURRENT_USER____STRING,
+        Fn_CURRENT_DATE____DATE,
+    )
+
+    @PartiQLFunctionExperimental
+    internal val aggregations = listOf(
+        Agg_EVERY__BOOL__BOOL,
+        Agg_ANY__BOOL__BOOL,
+        Agg_SOME__BOOL__BOOL,
+        Agg_COUNT__ANY__INT32,
+        Agg_COUNT_STAR____INT32,
+        Agg_MIN__INT8__INT8,
+        Agg_MIN__INT16__INT16,
+        Agg_MIN__INT32__INT32,
+        Agg_MIN__INT64__INT64,
+        Agg_MIN__INT__INT,
+        Agg_MIN__DECIMAL_ARBITRARY__DECIMAL_ARBITRARY,
+        Agg_MIN__FLOAT32__FLOAT32,
+        Agg_MIN__FLOAT64__FLOAT64,
+        Agg_MAX__INT8__INT8,
+        Agg_MAX__INT16__INT16,
+        Agg_MAX__INT32__INT32,
+        Agg_MAX__INT64__INT64,
+        Agg_MAX__INT__INT,
+        Agg_MAX__DECIMAL_ARBITRARY__DECIMAL_ARBITRARY,
+        Agg_MAX__FLOAT32__FLOAT32,
+        Agg_MAX__FLOAT64__FLOAT64,
+        Agg_SUM__INT8__INT8,
+        Agg_SUM__INT16__INT16,
+        Agg_SUM__INT32__INT32,
+        Agg_SUM__INT64__INT64,
+        Agg_SUM__INT__INT,
+        Agg_SUM__DECIMAL_ARBITRARY__DECIMAL_ARBITRARY,
+        Agg_SUM__FLOAT32__FLOAT32,
+        Agg_SUM__FLOAT64__FLOAT64,
+        Agg_AVG__INT8__INT8,
+        Agg_AVG__INT16__INT16,
+        Agg_AVG__INT32__INT32,
+        Agg_AVG__INT64__INT64,
+        Agg_AVG__INT__INT,
+        Agg_AVG__DECIMAL_ARBITRARY__DECIMAL_ARBITRARY,
+        Agg_AVG__FLOAT32__FLOAT32,
+        Agg_AVG__FLOAT64__FLOAT64,
+    )
+
+    @PartiQLFunctionExperimental
+    override val functions: List<PartiQLFunction> = scalars + operators + aggregations
```

Comment:
Had a offline discussion with this. Maybe consider move functions outside of plugin interface when we merge eval to main.

Reply:
IMO, it seems to me that the most logical place to put it would be in `ConnectorBindings` as the `bindings environment` really is just representative of the SQL environment (catalog and schema). It holds unique qualified names that represent some aspect of the environment (values/tables, views, functions, etc).

Reply:
Actually, I stand corrected on the uniqueness of names with regards to functions. Realistically, according to SQL:1999, all routines are placed in the `INFOMRATION_SCHEMA`'s `ROUTINES` table. Therefore, it really is separate from the bindings environment. The only relevant binding would be `<catalog>."INFORMATION_SCHEMA"."ROUTINES"` -- which we aren't getting into. Anyways, I've convinced myself that something separate from the `ConnectorBindings` would be adequate.

Reply:
Feedback reflected in interfaces draft here https://github.com/partiql/partiql-lang-kotlin/pull/1332